src/extensions/default/AutoUpdate/main.js

```diff
@@ -354,23 +397,48 @@ define(function (require, exports, module) {
 
         initializeState()
             .done(function () {
-                var initNodeFn = function () {
-                    postMessageToNode(MessageIds.INITIALIZE_STATE, _updateParams);
+
+                 var setUpdateInProgress = function() {
+                    var initNodeFn = function () {
+                        postMessageToNode(MessageIds.INITIALIZE_STATE, _updateParams, domainID);
+                    };
+
+                    if (updateJsonHandler.get('latestBuildNumber') !== _updateParams.latestBuildNumber) {
+                        setUpdateStateInJSON('latestBuildNumber', _updateParams.latestBuildNumber)
+                            .done(initNodeFn);
+                    } else {
+                        initNodeFn();
+                    }
                 };
 
-                if (updateJsonHandler.get('latestBuildNumber') !== _updateParams.latestBuildNumber) {
-                    setUpdateStateInJSON('latestBuildNumber', _updateParams.latestBuildNumber)
-                        .done(initNodeFn);
-                } else {
-                    initNodeFn();
-                }
+                checkIfAnotherSessionInProgress()
+                    .done(function(inProgress) {
+                        if(inProgress) {
+                            UpdateInfoBar.showUpdateBar({
+                                type: "error",
+                                title: Strings.AUTOUPDATE_ERROR,
+                                description: Strings.AUTOUPDATE_IN_PROGRESS
+                            });
+                        } else {
+                             setUpdateStateInJSON("autoUpdateInProgress", true)
+                                .done(setUpdateInProgress);
+
+                        }
+                    })
+                    .fail(function() {
+                        setUpdateStateInJSON("autoUpdateInProgress", true)
```

Comment:
If we fail to read the status of the session, should we be trying to start a session ? I would rather bail auto-update all together ?

Reply:
Mainly It will fail to read status of the session when File is not available or Key is not present in File.
in such cases we should try starting a session.